server/mailet/integration-testing/src/test/java/org/apache/james/transport/mailets/PerRecipientRateLimitMailetIntegrationTest.java

```diff
@@ -0,0 +1,311 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PerRecipientRateLimitMailetIntegrationTest {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PerRecipientRateLimitMailetIntegrationTest.class);
+    private static final int SEARCH_LIMIT_DEFAULT = 100;
+    private static final String SENDER = "sender@" + DEFAULT_DOMAIN;
+    private static final String SENDER2 = "sender2@" + DEFAULT_DOMAIN;
+    private static final String RECIPIENT = "recipient@" + DEFAULT_DOMAIN;
+    private static final String RECIPIENT2 = "recipient2@" + DEFAULT_DOMAIN;
+
+    private TemporaryJamesServer jamesServer;
+    private MailboxProbeImpl mailboxProbe;
+
+    @RegisterExtension
+    public TestIMAPClient imapClient = new TestIMAPClient();
+    @RegisterExtension
+    public SMTPMessageSender messageSender = new SMTPMessageSender(DEFAULT_DOMAIN);
+
+    @BeforeEach
+    void setup(@TempDir File temporaryFolder) throws Exception {
+        MailetContainer.Builder mailetContainer = TemporaryJamesServer.simpleMailetContainerConfiguration()
+            .putProcessor(ProcessorConfiguration.error()
+                .enableJmx(false)
+                .addMailet(MailetConfiguration.builder()
+                    .matcher(All.class)
+                    .mailet(ToRepository.class)
+                    .addProperty("repositoryPath", ERROR_REPOSITORY.asString()))
+                .build())
+            .putProcessor(ProcessorConfiguration.transport()
+                .addMailet(MailetConfiguration.builder()
+                    .matcher(All.class)
+                    .mailet(PerRecipientRateLimitMailet.class)
+                    .addProperty("duration", "60s")
+                    .addProperty("count", "1")
+                    .addProperty("size", "6K")
+                    .build())
+                .addMailetsFrom(CommonProcessors.transport()));
+
+        jamesServer = TemporaryJamesServer.builder()
+            .withMailetContainer(mailetContainer)
+            .withOverrides(new MemoryRateLimiterModule())
+            .build(temporaryFolder);
+        jamesServer.start();
+
+        jamesServer.getProbe(DataProbeImpl.class).fluent()
+            .addDomain(DEFAULT_DOMAIN)
+            .addUser(RECIPIENT, PASSWORD)
+            .addUser(RECIPIENT2, PASSWORD)
+            .addUser(SENDER, PASSWORD)
+            .addUser(SENDER2, PASSWORD);
+
+        mailboxProbe = jamesServer.getProbe(MailboxProbeImpl.class);
+    }
+
+    @AfterEach
+    void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    private void awaitFirstMessage() {
+        try {
+            imapClient.connect(LOCALHOST_IP, jamesServer.getProbe(ImapGuiceProbe.class).getImapPort())
+                .login(RECIPIENT, PASSWORD)
+                .select(TestIMAPClient.INBOX)
+                .awaitMessage(Constants.awaitAtMostOneMinute);
+        } catch (IOException e) {
+            LOGGER.error("Unexpected error", e);
+        }
```

Comment:
In a test class propagating exceptions instead of handling them is more than acceptable